ethcore/src/client/client.rs

```diff
@@ -2857,6 +2858,66 @@ impl IoChannelQueue {
 	}
 }
 
+impl PrometheusMetrics for Client {
+	fn prometheus_metrics(&self, r: &mut prometheus::Registry) {
+
+		// gas, tx & blocks
+		let report = self.report();
+
+		prometheus_counter(r, "import_gas", "Gas processed", report.gas_processed.as_u64() as i64);
+		prometheus_counter(r, "import_blocks", "Blocks imported", report.blocks_imported as i64);
+		prometheus_counter(r, "import_txs", "Transactions applied", report.transactions_applied as i64);
+
+		let state_db = self.state_db.read();
+		prometheus_gauge(r, "statedb_mem_used", "State DB memory used", state_db.mem_used() as i64);
+		prometheus_gauge(r, "statedb_cache_size", "State DB cache size", state_db.cache_size() as i64);
```

Comment:
Maybe drop the read-lock here?

Reply:
Yeah consider putting this under a scope `{ }`.